### PR TITLE
Added time stamps and time to live

### PR DESCRIPTION
**Pull request by robertbennett1998**

### CODE REVIEW

File: Microsoft.Azure.CosmosRepository/src/FullItem.cs [hunk 1]
```diff
@@ -31,15 +32,39 @@ namespace Microsoft.Azure.CosmosRepository
...
+                if (value?.TotalSeconds != null) _timeToLive = (int)value?.TotalSeconds;
+            }
+        }
+
+        [JsonProperty("ttl", NullValueHandling = NullValueHandling.Ignore)]
+        private int? _timeToLive;
+
+        /// <inheritdoc />
+        [JsonIgnore]
+        public DateTime LastUpdatedTimeUtc => new DateTime(1970, 1, 1, 0, 0, 0).AddSeconds(LastUpdatedTimeRaw);
```
Comment (mumby0168) [inline Microsoft.Azure.CosmosRepository/src/FullItem.cs:56]: We could extract this to a static readonly field, saves it been created new everytime, what do you think?

File: Microsoft.Azure.CosmosRepository/src/TimeStampedItem.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    /// </example>
+    public class TimeStampedItem : Item, IItemWithTimeStamps
+    {
+        /// <inheritdoc />
+        [JsonProperty]
+        public DateTime? CreatedTimeUtc { get; set; }
+
+        /// <inheritdoc />
+        [JsonIgnore]
+        public DateTime LastUpdatedTimeUtc => new DateTime(1970, 1, 1, 0, 0, 0).AddSeconds(LastUpdatedTimeRaw);
```
Comment (mumby0168) [inline Microsoft.Azure.CosmosRepository/src/TimeStampedItem.cs:40]: Same here about this, could be a static field within the library?

File: Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs [hunk 3]
```diff
@@ -24,16 +24,22 @@ namespace Microsoft.Azure.CosmosRepository
     /// <inheritdoc/>
     internal class InMemoryRepository<TItem> : IRepository<TItem> where TItem : class, IItem
     {
+        internal int CurrentTs => (int)(DateTime.UtcNow - new DateTime(1970, 1, 1, 0, 0, 0)).TotalSeconds;
```
Comment (mumby0168) [inline Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs:27]: Again could be static constant in the lib

Comment (robertbennett1998) [reply Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs:27]: Yeah that is a good idea! Will have a look now.

**Comment by mumby0168** [comment]
Only one thing on top of this @robertbennett1998 it can be a joint effort, but we could at least do with some new README sections for the etags, ttl & date features

**Comment by mumby0168** [comment]
> Only one thing on top of this @robertbennett1998 it can be a joint effort, but we could at least do with some new README sections for the etags, ttl & date features

We have plans for a docs site but for now these would be useful

**Review by IEvangelist — APPROVED** [review]
Looks good to me, we'll :shipit: - thank you 🙏

**Pull request merged**